### PR TITLE
added regression algorithm

### LINKED ISSUE #1583: [Idea]: Regression
### Idea Title

Regression

### Idea Description

Regression is a type of supervised learning algorithm used to predict a continuous outcome variable based on one or more predictor variables. It aims to model the relationship between the dependent variable (the outcome) and independent variables (the predictors) by fitting a mathematical function to the observed data. Regression analysis is widely applied in various fields, including finance, economics, and natural sciences, for tasks such as forecasting and risk assessment.

@ajay-dhangar kindly assign me this issue with the labels

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [X] Improves code quality
- [X] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by KGupta2601**
### Description
added regression algorithm

Fixes #1583 

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by KGupta2601** [comment]
@ajay-dhangar kindly review my PR and provide the labels of gssoc extended and level 1

File: docs/machine-learning/regression_algorithm.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
+---
+id: regression  
+title: Regression Algorithm (Supervised learning)
```
Comment (ajay-dhangar) [inline docs/machine-learning/regression_algorithm.md]: replace `title: Regression Algorithm (Supervised learning)` with `title: "Regression Algorithm (Supervised learning)"`

**Review by ajay-dhangar — APPROVED** [review]
Update your title only

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
follow best practices  for our docs

**Comment by KGupta2601** [comment]
@ajay-dhangar kindly check now...I have made the changes

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**